src/i18n/TextCollection/i18nTextCollector.php

```diff
@@ -548,12 +549,37 @@ public function collectFromCode($content, $module)
         $tokens = token_get_all("<?php\n" . $content);
         $inTransFn = false;
         $inConcat = false;
+        $inNamespace = false;
+        $inClass = false;
         $inArrayClosedBy = false; // Set to the expected closing token, or false if not in array
         $currentEntity = array();
+        $currentClass = []; // Class components
         foreach ($tokens as $token) {
             if (is_array($token)) {
                 list($id, $text) = $token;
 
+                // Check class
```

Comment:
I'll add to 4.0.0.md. This is so that you can use `_t(__CLASS__.'.SOMEKEY')` instead of the FQN.